src/postgres/backend/postmaster/peloton.cpp

```diff
@@ -166,42 +161,59 @@ peloton_ddl(Node *parsetree) {
  *  Handle DML requests in Peloton.
  * ----------
  */
-void
-peloton_dml(const PlanState *planstate,
-            bool sendTuples,
-            DestReceiver *dest,
-            TupleDesc tuple_desc,
-            const char *prepStmtName) {
+void peloton_dml(const PlanState *planstate, bool sendTuples,
+                 DestReceiver *dest, TupleDesc tuple_desc,
+                 const char *prepStmtName) {
   peloton_status status;
 
   // Get the parameter list
   assert(planstate != NULL);
   assert(planstate->state != NULL);
   auto param_list = planstate->state->es_param_list_info;
+  SubPlanState *subplanstate = nullptr;
+  if (planstate->state->es_param_exec_vals != nullptr)
+    subplanstate = reinterpret_cast<SubPlanState *>(
+        planstate->state->es_param_exec_vals->execPlan);
 
   // Create the raw planstate info
   std::shared_ptr<const peloton::planner::AbstractPlan> mapped_plan_ptr;
+  std::shared_ptr<const peloton::planner::AbstractPlan> mapped_subplan_ptr;
 
   // Get our plan
   if (prepStmtName) {
-    mapped_plan_ptr = peloton::bridge::PlanTransformer::GetInstance().GetCachedPlan(prepStmtName);
+    mapped_plan_ptr =
+        peloton::bridge::PlanTransformer::GetInstance().GetCachedPlan(
+            prepStmtName);
   }
 
   /* A cache miss or an unnamed plan */
   if (mapped_plan_ptr.get() == nullptr) {
-    auto plan_state = peloton::bridge::DMLUtils::peloton_prepare_data(planstate);
-    mapped_plan_ptr = peloton::bridge::PlanTransformer::GetInstance().TransformPlan(plan_state, prepStmtName);
+    auto plan_state =
+        peloton::bridge::DMLUtils::peloton_prepare_data(planstate);
+    mapped_plan_ptr =
+        peloton::bridge::PlanTransformer::GetInstance().TransformPlan(
+            plan_state, prepStmtName);
   }
 
   // Construct params
-  std::vector<peloton::Value> param_values = peloton::bridge::PlanTransformer::BuildParams(param_list);
-
-  //===----------------------------------------------------------------------===//
-  //   End for sending query
-  //===----------------------------------------------------------------------===//
+  std::vector<peloton::Value> param_values =
+      peloton::bridge::PlanTransformer::BuildParams(param_list);
```

Comment:
Can we move the subsequent code to `peloton::bridge::PlanTransformer::BuildParams` ? Like pass in another argument to it, and do this processing within that function.